### PR TITLE
Fix offset of extended argument instruction

### PR DESCRIPTION
**Pull request by tangboxuan**
There is currently a bug in this file when parsing some JUMP instructions to a EXTENDED_ARG instruction. From what I understand, EXTENDED_ARG instructions are combined with the next instruction into one offset with the format OFFSET1_OFFSET2. This throws an error in this part of code by triggering a string and integer comparison.

Sample bytecode from dis.dis: (Evilnum Pyvil RAT MD5:12efa6d0f23346f2a6081969528839cc8712676f8f1e3658b1d15cd4bd7d3b5b de-obfuscated)

```
204        1112 LOAD_NAME              171 (O7)
           1114 EXTENDED_ARG             1
           1116 LOAD_NAME              258 (i)
           1118 BINARY_SUBSCR
           1120 LOAD_NAME              195 (f6)
           1122 LOAD_NAME              236 (j)
           1124 BINARY_SUBSCR
           1126 BINARY_XOR
           1128 LOAD_NAME              225 (k5)
           1130 EXTENDED_ARG             1
           1132 LOAD_NAME              258 (i)
           1134 STORE_SUBSCR
           1136 JUMP_FORWARD            82 (to 1220)

206     >> 1138 LOAD_NAME               20 (slice)
           1140 EXTENDED_ARG             1
           1142 STORE_NAME             278 (D2)

207        1144 LOAD_NAME              243 (super)
           1146 EXTENDED_ARG             1
           1148 STORE_NAME             279 (v1)

208        1150 LOAD_NAME              199 (object)
           1152 EXTENDED_ARG             1
           1154 STORE_NAME             280 (e0)

209        1156 LOAD_NAME               10 (str)
           1158 LOAD_ATTR               45 (count)
           1160 EXTENDED_ARG             1
           1162 STORE_NAME             281 (w7)

210        1164 LOAD_NAME              126 (format)
           1166 EXTENDED_ARG             1
           1168 STORE_NAME             282 (g6)

211        1170 LOAD_NAME              131 (frozenset)
           1172 EXTENDED_ARG             1
           1174 STORE_NAME             283 (A1)

212        1176 LOAD_NAME               83 (map)
           1178 EXTENDED_ARG             1
           1180 STORE_NAME             284 (Q7)

213        1182 LOAD_NAME               27 (hex)
           1184 EXTENDED_ARG             1
           1186 STORE_NAME             285 (S4)

214        1188 EXTENDED_ARG             1
           1190 LOAD_NAME              286 (abs)
           1192 EXTENDED_ARG             1
           1194 STORE_NAME             287 (s9)

215        1196 LOAD_NAME               10 (str)
           1198 LOAD_ATTR               17 (splitlines)
           1200 EXTENDED_ARG             1
           1202 STORE_NAME             288 (Q5)

216        1204 LOAD_NAME              171 (O7)
           1206 EXTENDED_ARG             1
           1208 LOAD_NAME              258 (i)
           1210 BINARY_SUBSCR
           1212 LOAD_NAME              225 (k5)
           1214 EXTENDED_ARG             1
           1216 LOAD_NAME              258 (i)
           1218 STORE_SUBSCR

217     >> 1220 LOAD_NAME               10 (str)
           1222 EXTENDED_ARG             1
           1224 STORE_NAME             289 (w3)
```

This fix seems to work on some of the samples that I have tried in Python 3.7, unsure of how it will affect Python 2.

### CODE REVIEW

File: uncompyle6/parsers/reducecheck/ifelsestmt.py [hunk 1]
```diff
@@ -158,13 +158,15 @@ def ifelsestmt(self, lhs, n, rule, tree, tokens, first, last):
     # just the last one.
     if len(tree) == 5:
         end_come_froms = tree[-1]
+        offset = tokens[first].offset
+        offset = int(tokens[first].offset.split('_')[0]) if isinstance(offset, str) else offset
```
Comment (rocky) [inline uncompyle6/parsers/reducecheck/ifelsestmt.py:162]: Please off off2int here.

**Comment by rocky** [comment]
Thanks. Please add a test for this.  Take something that before this patch does not decompile and then after then decompiles.

Strip that down into the smallest program that does this and the compile it and put it in `test/bytecode_x.x` as appropriate.
`python-uncompyle6/test/simple_source/bug36/01_extended_arg.py` is an example of a test that needs to make use of `EXTENDED_ARG`. 

Best is to craft that test program so that when it is run in Python, it passes assert statements in the control flow places it is supposed to pass and raises an error if the control flow isn't right. 

`python-uncompyle6/test/simple_source/bug36/10_fstring.py` is an example of a runnable test.

See https://github.com/rocky/python-uncompyle6/wiki/Adding-Tests for how to add a test.

**Comment by tangboxuan** [comment]
Not sure if there is a bug in bin/uncompyle6 (which add-test.py calls) since "from uncompyle6.bin.uncompile import main_bin" imports the pip module rather than the local module for me (Python 3.7 on Mac OS).

On uninstalling uncompyle6 I still get this:
Traceback (most recent call last):
  File "../bin/uncompyle6", line 2, in <module>
    from uncompyle6.bin.uncompile import main_bin
ModuleNotFoundError: No module named 'uncompyle6'

**Comment by rocky** [comment]
> Not sure if there is a bug in bin/uncompyle6 (which add-test.py calls) since "from uncompyle6.bin.uncompile import main_bin" imports the pip module rather than the local module for me (Python 3.7 on Mac OS).
> 
> On uninstalling uncompyle6 I still get this: Traceback (most recent call last): File "../bin/uncompyle6", line 2, in from uncompyle6.bin.uncompile import main_bin ModuleNotFoundError: No module named 'uncompyle6'

Pretty sure this is something in your environment. Try running the program with an explicit python interpreter.

**Comment by rocky** [comment]
> This fix seems to work on some of the samples that I have tried in Python 3.7, unsure of how it will affect Python 2.

Probably a problem there as well. In Python2 this is less noticeable because offsets can be 3 bytes.

BTW this fix probably also needs to be applied to python3-decompile.

**Review by rocky — APPROVED** [review]
LGTM - thanks or seeing this through!

**Pull request merged**